lib/astutils.cpp

```diff
@@ -1467,7 +1467,7 @@ bool isUsedAsBool(const Token* const tok, const Settings* settings)
     if (Token::Match(parent, "&&|!|%oror%"))
         return true;
     if (parent->isCast())
-        return isUsedAsBool(parent);
+        return !Token::simpleMatch(parent->astOperand1(), "dynamic_cast") && isUsedAsBool(parent);
```

Comment:
Its still used as a bool even with `dynamic_cast`. This looks like this should be fixed in ValueFlow. We shouldn't have a known value when using a `dynamic_cast`.

Reply:
Ah nevermind, its not directly used as a bool here. The ValueFlow is correct.